### PR TITLE
Fix the slf4j binding issue

### LINKED ISSUE #2020: Log4j2 properties file not working for apim-analytics-publisher related logs
### Description:

https://github.com/wso2/apim-analytics-publisher is used as a dependency for the microgateway to publish analytics data. When we try to debug that component alone using log4j2.properties file, the log level is not applied. The issue was due to sl4j  binding is not properly done.

### Steps to reproduce:
enable analytics in config.toml
Set the following to the log4j2.properties file.

```
logger.enforcer.name = org.wso2.am.analytics.publisher.client
logger.enforcer.level = debug
logger.enforcer.additivity = false
logger.enforcer.appenderRef.rolling.ref = ENFORCER_LOGFILE
logger.enforcer.appenderRef.console.ref = ENFORCER_CONSOLE

```

### Affected Product Version:

Affected/Choreo Connect 0.9.0.3

### Environment details (with versions):
- OS: MacOS
- Client:
- Env (Docker/K8s): Docker

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
log4j2 properties file is not correctly applied for the components from https://github.com/wso2/apim-analytics-publisher. The reason is that slf4j binding is not done properly.

### Issues

Fixes #2020

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Docker 

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer/pom.xml [hunk 1]
```diff
@@ -140,6 +140,8 @@
                         <excludes>
                             <exclude>org.apache.logging:*</exclude>
                             <exclude>log4j:*</exclude>
+                            <exclude>org.slf4j:slf4j-log4j12</exclude>
+                            <exclude>org.apache.logging.log4j:log4j-slf4j-impl</exclude>
```
Comment (Rajith90) [inline enforcer/pom.xml:144]: Doesn't this captured in the `<exclude>org.apache.logging:*</exclude>` statement

File: enforcer/pom.xml [hunk 2]
```diff
@@ -387,6 +385,11 @@
             <artifactId>Java-WebSocket</artifactId>
             <version>1.5.1</version>
         </dependency>
-
+        <!-- slf4j binding -->
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j-impl</artifactId>
+            <version>2.13.3</version>
```
Comment (Rajith90) [inline enforcer/pom.xml:392]: Lets move the version to root pom

File: enforcer/pom.xml (hunk 1 above)
Comment (VirajSalaka) [reply enforcer/pom.xml:144]: No
The reason is that the group ID s are different

File: enforcer/pom.xml (hunk 2 above)
Comment (VirajSalaka) [reply enforcer/pom.xml:392]: my bad.
Will fix

Comment (VirajSalaka) [reply enforcer/pom.xml]: fixed via https://github.com/wso2/product-microgateway/pull/2021/commits/8ef5c052d366170ddbee991a9d2b752ee291e294

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**